Src/BootCamp.Chapter/Csv/CsvBase.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BootCamp.Chapter.Csv
+{
+    public class CsvBase
+    {
+        public CsvRow Header { get; protected set; } = new CsvRow();
+        public string Footer { get; protected set; } = string.Empty;
+        public CsvDelimiter Delimiter { get; protected set; } = CsvDelimiter.Comma;
+        protected string FileName { get; set; }
+        protected bool HasHeader { get; set; }
+        protected bool HasFooter { get; set; }
+        protected IList<CsvRow> Rows { get; set; } = new List<CsvRow>();
+
+        public CsvBase(string fileName)
+        {
+            if (!fileName.IsValid())
+            {
+                throw new ArgumentException("fileName cannot be null or empty");
+            }
+
+            FileName = fileName;
+        }
+
+        public CsvBase(string fileName, CsvDelimiter delimiter, bool hasHeader)
+        {
+            if (!fileName.IsValid())
+            {
+                throw new ArgumentException("fileName cannot be null or empty");
+            }
+
+            FileName = fileName;
+            Delimiter = delimiter;
+            HasHeader = hasHeader;
+        }
+
+        public CsvBase(string fileName, CsvDelimiter delimiter, bool hasHeader, bool hasFooter)
```

Comment:
Remove the first ctor and instead have optional 4th parameter with a default value...
```cs
public CsvBase(string fileName, CsvDelimiter delimiter, bool hasHeader, bool hasFooter = default)
{
  ...
}
```

Reply:
That's how I did it the first time then I read on codacy test we should not use optional parameters in constructors

Reply:
I find that hard to justify. Many things have a right spot. If you can have a ctor with default parameter, what wrong with that?